### PR TITLE
Add Sentry to the Foundation site

### LINKED ISSUE #3874: Add Sentry to the foundation site
Rather than an overwhelming "let's add testing", we should start simple: Let's add Sentry support to the Foundation site, so we can catch errors around petitions and the buyer's guide.

### PR DESCRIPTION
**Pull request by patjouk**
Closes #3874 

@cadecairos did all the work on the `donate-wagtail` so it was super easy to set up here :)
Front and back tested: errors are visible on sentry in the resolved section.

I'm still waiting on MoCo to add the rest of the team on Sentry and create the prod app, so let's wait next week to merge this PR.

ToDos:
- [x] When sentry prod is ready, add the prod DSN var to Heroku,
- [x] Check that the staging is properly configured on Sentry and that we're filtering any sensitive info,
- [x] Same, but with the prod project.

### CODE REVIEW

**Comment by patjouk** [comment]
I found those fields that should be scrub in Sentry:
`givenNames`, `surname`, `country`, `lang`, `email`, `comment`, `postalCode`. It's mostly petitions and newsletter sing up. Am I forgetting something obvious @alanmoo @Pomax @cadecairos?

File: network-api/networkapi/settings.py [hunk 1]
```diff
@@ -85,6 +88,18 @@
...
+SENTRY_DSN = env('SENTRY_DSN')
+HEROKU_RELEASE_VERSION = env('HEROKU_RELEASE_VERSION')
+
+if SENTRY_DSN:
+    import sentry_sdk
+    from sentry_sdk.integrations.django import DjangoIntegration
+    sentry_sdk.init(
+        dsn=SENTRY_DSN,
+        integrations=[DjangoIntegration()],
+        release=HEROKU_RELEASE_VERSION
```
Comment (cadecairos) [inline network-api/networkapi/settings.py]: We'll need to enable the Dyno Metadata feature on staging and production to make this variable available to use: https://devcenter.heroku.com/articles/dyno-metadata

File: source/js/main.js [hunk 2]
```diff
@@ -20,6 +21,9 @@ import youTubeRegretsTunnel from "./youtube-regrets.js";
 
 const SHOW_MEMBER_NOTICE = false;
 
+// Initialize Sentry error reporting
+Sentry.init({ dsn: __SENTRY_DSN__ });
```
Comment (cadecairos) [inline source/js/main.js]: any way to get the `HEROKU_RELEASE_VERSION` variable rendering client-side?

Comment (cadecairos) [reply source/js/main.js]: Aha, below we have a request to fetch the client-side config variables, lets add the version number to that and move the sentry init to that code block.

Comment (cadecairos) [reply source/js/main.js]: oh wait, nevermind, you can just use the same define plugin method you already used.

File: network-api/networkapi/settings.py (hunk 1 above)
Comment (patjouk) [reply network-api/networkapi/settings.py]: Done for both staging and prod.

File: source/js/main.js (hunk 2 above)
Comment (cadecairos) [inline source/js/main.js]:
```suggestion
Sentry.init({
  dsn: __SENTRY_DSN__,
  release: __HEROKU_RELEASE_VERSION__
});
```

**Comment by cadecairos** [comment]
@patjouk this isn't a draft PR anymore, right?

**Comment by patjouk** [comment]
I kept it as a draft because I'm still waiting from MoCo to create the prod app and add new people to our team on Sentry. But like you said in a previous meeting, we can merge it for staging and finish it for prod later.

**Comment by patjouk** [comment]
I created the prod app and will ask for an update about my ticket on bugzilla. Meanwhile, let's merge this.

**Pull request merged**